src/relay.ts

```diff
@@ -151,6 +151,35 @@ async function isContract(address: string): Promise<boolean> {
   return isContract;
 }
 
+async function submitTx(
+  relayerContract: RelayerContract,
+  isRetryAttempt: boolean,
+  logger: Logger,
+): Promise<boolean> {
+  await relayerContract.simulate.relay();
+
+  const baseGasFee = parseGwei("25");
+  // Attempt to use 2x the base gas fee if it's a retry attempt, otherwise let the RPC decide
+  const params = isRetryAttempt
+    ? { maxFeePerGas: baseGasFee * 2n, maxPriorityFeePerGas: baseGasFee * 2n }
+    : {};
+
+  // @ts-expect-error todo: tricky to get the params recognized when using the type from the abi
+  const hash = await relayerContract.write.relay(params);
```

Comment:
I think this is a legitimate typescript error

I think the general viem syntax for calling smart contract functions is `contract.write.functionName(args, options)`

And you're passing the args as options here, so I think this would be the correct syntax (although the viem docs are surprisingly unclear around using `write` over `writeContract`)

```suggestion
  const hash = await relayerContract.write.relay([], params);
```